src/OpenApi/Model/Parameter.php

```diff
@@ -55,12 +55,12 @@ public function getDeprecated(): bool
 
     public function canAllowEmptyValue(): ?bool
     {
-        return $this->allowEmptyValue;
+        return 'query' === $this->in ? $this->allowEmptyValue : null;
```

Comment:
actually in the constructor if it's not a `path` parameter we set $this->allowEmptyValue don't change in the getters/setters